src/thread_task.rs

```diff
@@ -1090,26 +1077,26 @@ pub unsafe extern "C" fn rav1d_worker_task(data: *mut c_void) -> *mut c_void {
                                 if res_1.is_err() {
                                     pthread_mutex_lock(&mut (*ttd).lock);
                                     // memory allocation failed
-                                    (*f).task_thread.done[(2 - p_0) as usize]
+                                    f.task_thread.done[(2 - p_0) as usize]
                                         .store(1 as c_int, Ordering::SeqCst);
-                                    (*f).task_thread
-                                        .error
-                                        .store(-(1 as c_int), Ordering::SeqCst);
-                                    (*f).task_thread.task_counter.fetch_sub(
-                                        frame_hdr.tiling.cols * frame_hdr.tiling.rows + (*f).sbh,
+                                    f.task_thread.error.store(-(1 as c_int), Ordering::SeqCst);
+                                    let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
```

Comment:
Can you hoist these unwraps as high as possible so we do them as few times as possible?  I had put them close to their uses previously since the control flow was very complex, but now it's a lot simpler and clear.

Reply:
Nope, overlapping borrows. I had to move them down to their uses in some places.

Reply:
Are the overlapping borrows from other borrows of these specific fields or some function that takes `&mut f`?